### DISCUSSION TITLE
Squishing my commits

### DISCUSSION THREAD

**Discussion by JavaJoeS** [eclipse-platform/eclipse.platform#1229] [General]
Been using git for some number of years, Squishing down commits is new to me.  I found tons of examples all over, but no recommendations in our Eclipse Foundation docs.   Which is preferred;  using Eclipse or command line?  What are "our" steps
that we need to follow?

**Comment by fedejeanne** [comment]
This is probably cleared by now but my 2 cents on the issue are:
## Option 1: Amend + Force-Push
If you currently only have 1 commit in your branch, you can **amend** it
![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/40751244-2df9-48bf-821b-ab85df8f3687)

... and then you force-push the commit to your PR-branch

![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/747e8594-14de-4337-b1b8-961daebc8f83)

## Option 2: Interactive rebase + Force-Push
If you have several commits that you want to squash together then you do an **interactive rebase** on `master`, pick the 1st commit and squash the rest

![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/fa75a46c-424f-43c7-b2e7-8c6bd1bad923)

... and then you force-push the commit to your PR-branch

Does this answer your question?

**Reply by HeikoKlare** [reply]
Just one nitpicky remark: Force pushing is _always_ necessary when changing commits that have already been pushed, no matter if its via an amendment or via a rebase, so for both option 1 and 2.
